WooCommerce/Classes/POS/Presentation/Card Present Payments/Connection Alerts/PointOfSaleCardPresentPaymentUpdatingReaderAlertViewModel.swift

```diff
@@ -0,0 +1,77 @@
+import Foundation
+import SwiftUI
+
+struct PointOfSaleCardPresentPaymentUpdatingReaderAlertViewModel {
+    let title: String
+    let image: Image
+    let progressTitle: String
+    let progressSubtitle: String?
+    let cancelButtonTitle: String
+    let cancelReaderUpdate: (() -> Void)?
+
+    init(requiredUpdate: Bool, progress: Float, cancel: (() -> Void)?) {
+        self.image = Image(uiImage: .softwareUpdateProgress(progress: CGFloat(progress)))
+
+        let isComplete = progress == 1
+
+        self.title = isComplete ? Localization.titleComplete : Localization.title
+        self.progressTitle = String(format: Localization.percentCompleteFormat, 100 * progress)
+        self.progressSubtitle = isComplete ? nil : (requiredUpdate ? Localization.messageRequired : Localization.messageOptional)
+
+        self.cancelButtonTitle = isComplete ? Localization.dismissButtonText :
+        (requiredUpdate ? Localization.cancelRequiredButtonText : Localization.cancelOptionalButtonText)
+        self.cancelReaderUpdate = cancel
+    }
+}
+
+private extension PointOfSaleCardPresentPaymentUpdatingReaderAlertViewModel {
+    enum Localization {
+        static let title = NSLocalizedString(
+            "pointOfSale.cardPresentPayment.alert.updatingReader.title",
+            value: "Updating software",
+            comment: "Dialog title that displays when a software update is being installed"
+        )
+
+        static let titleComplete = NSLocalizedString(
+            "pointOfSale.cardPresentPayment.alert.updatingReader.title",
+            value: "Software updated",
+            comment: "Dialog title that displays when a software update just finished installing"
+        )
+
+        static let messageRequired = NSLocalizedString(
+            "pointOfSale.cardPresentPayment.alert.updatingReader.title",
+            value: "Your card reader software needs to be updated to collect payments. Cancelling will block your reader connection.",
+            comment: "Label that displays when a mandatory software update is happening"
+        )
+
+        static let messageOptional = NSLocalizedString(
+            "pointOfSale.cardPresentPayment.alert.updatingReader.title",
```

Comment:
The keys for these are all the same, but should be different. As is, I think it will mean that when translations are applied, only one string will be used in all three places.

Reply:
Oops my bad, I meant to have them all unique. Fixed in https://github.com/woocommerce/woocommerce-ios/pull/13085/commits/9fb3cfbb3f8af24f225934f77db3556b4ec4aca6.